planner/core/preprocess.go

```diff
@@ -459,6 +460,31 @@ func (p *preprocessor) checkCreateViewGrammar(stmt *ast.CreateViewStmt) {
 	}
 }
 
+func (p *preprocessor) checkCreateViewWithSelect(stmt *ast.SelectStmt) {
+	if stmt.SelectIntoOpt != nil {
+		p.err = ddl.ErrViewSelectClause.GenWithStackByArgs("INFO")
+		return
+	}
+	if stmt.LockTp != ast.SelectLockNone {
+		stmt.LockTp = ast.SelectLockNone
+		return
+	}
+}
+
+func (p *preprocessor) checkCreateViewWithSelectGrammar(stmt *ast.CreateViewStmt) {
+	switch stmt := stmt.Select.(type) {
+	case *ast.SelectStmt:
+		p.checkCreateViewWithSelect(stmt)
+	case *ast.UnionStmt:
```

Comment:
should check it in recursion?
```
create view as (A union select 1 into xxx) union C
```

Reply:
I think parser will convert sql `(A union select 1 into xxx) union C`  to `A union select 1 into xxx union C`, so we do not need to check here.

Reply:
cool